### PR TITLE
JAMES-2655 Simplify JPA RRT query logic

### PR DESCRIPTION
**Pull request by chibenwa**
Previous queries were complex and were syntactically incorrect when run against
MariaDB as reported by Marc Chamberlin.

This lead to portability issue.

Note that simplifying queries is an attempt to solve this.

### CODE REVIEW

File: server/data/data-jpa/src/main/java/org/apache/james/rrt/jpa/JPARecipientRewriteTable.java [hunk 1]
```diff
@@ -76,11 +76,16 @@ public void addMapping(MappingSource source, Mapping mapping) throws RecipientRe
...
     @Override
     protected Mappings mapAddress(String user, Domain domain) throws RecipientRewriteTableException {
-        Mappings mapping = getMapping(user, domain, "selectExactMappings");
-        if (!mapping.isEmpty()) {
-            return mapping;
+        Mappings userDomainMapping = getMapping(user, domain, "selectUserDomainMapping");
+        if (userDomainMapping != null && !userDomainMapping.isEmpty()) {
+            return userDomainMapping;
+        }
+        MappingSource domainDource = MappingSource.fromDomain(domain);
```
Comment (hoangdat) [inline server/data/data-jpa/src/main/java/org/apache/james/rrt/jpa/JPARecipientRewriteTable.java]: s/domainDource/domainSource

Comment (chibenwa) [reply server/data/data-jpa/src/main/java/org/apache/james/rrt/jpa/JPARecipientRewriteTable.java]: Héhé!

**Comment by chibenwa** [comment]
I added some little refactorings along the way...

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**